newIDE/app/src/InstancesEditor/SelectedInstances.js

```diff
@@ -75,7 +90,7 @@ export default class InstancesSelection {
     return this.pixiContainer;
   }
 
-  _renderButton(show, buttonObject, canvasPosition, size) {
+  _renderButton(show, buttonObject, canvasPosition, size, shape = 0) {
```

Comment:
Shape would benefit from being an "enum". JavaScript does not have "enums", but you can still mimic them like this:

Top of the file:
```
const RECTANGLE_BUTTON_SHAPE = 0;
const CIRCLE_BUTTON_SHAPE = 1;
```

(writing in ALL_CAPS is to denote constant variables that should not be changed at runtime, and is an hint that this is an enum too).
It's then more readable when calling _renderButton: 

```
this._renderButton(..., CIRCLE_BUTTON_SHAPE);
```

Finally, as _renderButton is only used in this file, I would go for making everything explicit and avoid the default argument for shape. Would be different if the method was used a lot everywhere in the code base, but that's not the case :)

(You could even add flow typing: at the top of the file `type ButtonShape = RECTANGLE_BUTTON_SHAPE|CIRCLE_BUTTON_SHAPE`, then `_renderButton(..., shape = RECTANGLE_BUTTON_SHAPE: ButtonShape)`. Of course the rest of the file is not flow typed, so there is more work).

Reply:
very good idea.It will help with code clarity!